### PR TITLE
Feature: support symlink for NodePublish

### PR DESCRIPTION
**Pull request by wangshli**
### Ⅰ. Describe what this PR does
support csi plugin use symlink for NodePublishVolum by add annotation in runtime, such as below:
- data.fluid.io/metadataList: "[{"Labels": {"fluid.io/node-puhlish-method": "symlink"}, "selector": { "kind": "PersistentVolume"}}]"

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/csi/plugins/nodeserver.go [hunk 1]
```diff
@@ -577,3 +592,38 @@ func (ns *nodeServer) prepareSessMgr(workDir string) error {
 
 	return nil
 }
+
+func createSymlink(targetPath, mountPath string) error {
```
Comment (cheyang) [inline pkg/csi/plugins/nodeserver.go:596]: Please move this function to utils pkg and create unit test for this change.

Comment (wangshli) [reply pkg/csi/plugins/nodeserver.go]: done.

File: pkg/csi/plugins/nodeserver.go [hunk 2]
```diff
@@ -577,3 +591,16 @@ func (ns *nodeServer) prepareSessMgr(workDir string) error {
 
 	return nil
 }
+
+func removeSymlink(targetPath string) (*csi.NodeUnpublishVolumeResponse, error) {
+	glog.Infof("remove symlink targetPath %v", targetPath)
+	if err := os.Remove(targetPath); err != nil && !os.IsNotExist(err) {
```
Comment (cheyang) [inline pkg/csi/plugins/nodeserver.go:597]: Can you use `unlink` ? We are trying to avoid `os.Remove`

File: pkg/csi/plugins/nodeserver.go [hunk 3]
```diff
@@ -577,3 +591,16 @@ func (ns *nodeServer) prepareSessMgr(workDir string) error {
 
 	return nil
 }
+
+func removeSymlink(targetPath string) (*csi.NodeUnpublishVolumeResponse, error) {
+	glog.Infof("remove symlink targetPath %v", targetPath)
```
Comment (cheyang) [inline pkg/csi/plugins/nodeserver.go:596]: I think this logic should determine if the path is symlink , then do the remove.

File: pkg/utils/symlink.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	// If the target path does not exist, it will be created when create symlink
+	if err != nil {
+		if !os.IsNotExist(err) {
+			glog.Errorf("Failed to lstat targetPath %s error %v", targetPath, err)
+			return status.Error(codes.Internal, err.Error())
+		}
+	} else {
+		// symlink would create targetPath so delete it first
+		glog.Infof("Deleting the targetPath before create symlink %v", targetPath)
+		err := os.Remove(targetPath)
```
Comment (cheyang) [inline pkg/utils/symlink.go:23]: When you delete the file, it requires very careful.

File: charts/fluid/fluid/values.yaml [hunk 5]
```diff
@@ -41,6 +41,8 @@ csi:
     rootDir: /var/lib/kubelet
   pruneFs: fuse.alluxio-fuse,fuse.jindofs-fuse,fuse.juicefs,fuse.goosefs-fuse,ossfs,alifuse.aliyun-alinas-efc
   recoverWarningThreshold: 50
+  # default method is "bind-mount", "symlink" is also support
+  nodePublishMethod: bind-mount
```
[the remark this answers is not in the document]
Comment (wangshli) [reply charts/fluid/fluid/values.yaml:45]: ok.

**Comment by zwwhdls** [comment]
Also need to check if `bindMount` or `symlink` before auto recovery.

**Comment by wangshli** [comment]
> Also need to check if `bindMount` or `symlink` before auto recovery.

ok.

File: pkg/csi/plugins/nodeserver.go (hunk 3 above)
Comment (wangshli) [reply pkg/csi/plugins/nodeserver.go]: done.

**Review by TrafalgarZZZ — COMMENTED** [review]
/lgtm

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**